src/content/peerconnection/endtoend-encryption/js/main.js

```diff
@@ -114,20 +120,26 @@ function encodeFunction(chunk, controller) {
 }
 
 function decodeFunction(chunk, controller) {
+  const view = new DataView(chunk.data);
+  const checksum = view.getUint32(chunk.data.byteLength - 4);
   if (currentCryptoKey) {
-    const view = new DataView(chunk.data);
-    const checksum = view.getUint32(chunk.data.byteLength - 4);
-    if (checksum != 0xDEADBEEF) {
+    if (checksum !== 0xDEADBEEF) {
```

Comment:
To make it more realistic: Have a counter "currentCryptoKeyGeneration" that is incremented every time you call setCryptoKey and add that to the checksum (0xDEADBEEF + currentCryptoKeyGeneration).
That way, frames encrypted with the wrong key will be discarded reliably, and changes from encrypt-with-key-A to encrypt-with-key-B will be handled, instead of just changes from encrypt-with-key-A to no-encrypt and vice versa.

Reply:
added as a separate commit. Single byte instead of the four bytes MKI from SRTP ;-)
The console.log shows up nicely when changing the key